Symplified financial system.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+contract UserManagement {
+
+    struct User {
+        string name;
+        string role;
+    }
+
+    mapping(address => User) private user;
+
+    address internal admin;
+    constructor() {
+        admin = msg.sender;
+    }
+    modifier onlyAdmin {
+        require(msg.sender == admin, "Only admin can  this function");
+        _;
+    }
+
+    function userExists() public view returns (bool) {
+        return bytes(user[msg.sender].name).length > 0;
+    }
+    function addUser(string memory name, string memory role) public onlyAdmin{
+        require(!userExists(), "User already exists!");
+
+        user[msg.sender] = User(name, role);
```

Comment:
Why do you want to replace information of admin :v 
How can i update role for a user :v

Reply:
function addUser(address userAddress, string memory name, string memory role) public onlyAdmin{
        require(!userExists(), "User already exists!");
        user[userAddress] = User(name, role);
    }
    
    function updateUserRole(address userAddress, string memory newRole) public onlyAdmin {
        require(userExists(), "User does not exist!");
        user[userAddress].role = newRole;
    }